lib/at_cmd_parser/include/at_params.h

```diff
@@ -0,0 +1,221 @@
+/**@file at_params.h
+ *
+ * @brief Store a list of AT commands/responses parameters.
+ *
+ * A parameter list contains an array of parameters defined by a type
+ * and a value. Those parameters could be arguments of an AT command for
+ * instance. They can be numeric or string values.
+ * The same list of parameters can be reused. Each parameter can be
+ * updated or cleared. Once the parameter list is created, its size
+ * cannot be changed. All parameters values are copied in the list.
+ * Parameters should be cleared to free that memory. Getter and setter
+ * methods are available to read parameter values.
+ *
+ */
+
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef AT_PARAMS_H_
+#define AT_PARAMS_H_
+
+#include <stdint.h>
+
+/**
+ * @brief List of AT parameters that compose an AT command or response.
+ *
+ * Contains an array of opaque data. Setter and getter methods should be used
+ * to get access to the parameters in the array.
+ */
+
+enum at_param_type {
+	AT_PARAM_TYPE_EMPTY,
```

Comment:
This value should explicitly be 0. This is the default value when initializing the element. This was marked as a comment in the original implementation.

Reply:
I don't agree, there are two reasons for this:

1. You should usually not care about the value an enumerator value has, if you do you should consider using constants or #defines instead.
2. The C standard (C99) clearly states that the first member in an enumerator has the value zero.